react/tic-tac-toe/src/app/components/AuthRoute/index.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Redirect, Route } from 'react-router';
+import { connect } from 'react-redux';
+import { bool } from 'prop-types';
+
+function AuthRoute({ component: Component, redirectPath, loginComponent, loggedIn, ...props }) {
+  let route = loggedIn ? <Route {...props} component={Component} /> : <Redirect to={redirectPath} />;
+  if (loginComponent) {
+    route = loggedIn ? <Redirect to={redirectPath} /> : <Route {...props} component={Component} />;
+  }
+  return route;
+}
+
+const mapStateToProps = state => ({
+  loggedIn: state.users.token !== null
+});
+
+AuthRoute.propTypes = {
+  loggedIn: bool
+};
```

Comment:
I think you are missing redirectPath prop here. Also, you should add default props